ably/realtime_client.go

```diff
@@ -72,16 +72,15 @@ func (c *RealtimeClient) onReconnectMsg(msg *proto.ProtocolMessage) {
 	switch msg.Action {
 	case proto.ActionConnected:
 		if c.Connection.id == msg.ConnectionID {
+			c.Connection.queue.Flush()
```

Comment:
Sorry, this is wrong now, we should only start processing the queue when the state is CONNECTED. It was right in that sense before.

If you look at [realtime_conn.go](https://github.com/ably/ably-go/blob/feature/RTN15b-c/ably/realtime_conn.go#L440-L457), when CONNECTED arrives while not reconnecting, there's a lot of logic that also must apply when we _are_ reconnecting. Some of that logic is replicated here.

I'd suggest that that block of code should be reached whether we're reconnecting or not. And then, only if we're reconnecting, we call `onReconnect(err)` in order to reattach channels and _only_ to reattach channels. The rest of the logic is common.

Reply:
That is very nice observation.

Reply:
I moved as much logic to `Conn.eventLoop` only changes touching channels remain on the call back https://github.com/ably/ably-go/pull/159/commits/9ac3c19fa787933298376d891ebb9796e97d60d8